vowpalwabbit/v_array.h

```diff
@@ -412,6 +412,12 @@ struct v_array<T, typename std::enable_if<std::is_trivially_copyable<T>::value>:
 
     return false;
   }
+
+  friend bool operator==(const v_array<T>& lhs, const v_array<T>& rhs)
```

Comment:
is this being used in this PR?

Reply:
the benchmarks for text parsing seem to be complaining a bit for this PR (cache looks good) so was wondering what it could be

Reply:
It's being used in the tests to compare v_arrays, but not in the library code anywhere (explicitly). I am not sure what could really be having that effect on the benchmarks as this PR is just a rearrange, adding consts and adding a test

Reply:
Just verified that this is not called in the benchmarks